test/app/features/appstate/domain/usecases/app_preferences_usecase_test.dart

```diff
@@ -148,7 +182,9 @@ void main() {
       );
 
       expect(value.isRight(), true);
-      expect(value.getOrElse(() => null), AppRoute('/authentication/stealth'));
+      value.forEach((route) {
+        expect(route, AppRoute('/authentication/stealth'));
+      });
```

Comment:
```suggestion
      expect(
        value.fold((l) => l, (r) => r),
        AppRoute('/authentication/stealth'),
      );
```

Acredito que utilizar o `.fold` facilita a leitura do teste.

Reply:
você tem total razão! por ser algo meio repetitivo, pensei em fazer uma extension para esses casos de testes, o que acha?
```dart
extension EitherExt<L, R> on Either<L, R> {
  R? getOrNull() => fold((l) => null, (r) => r);
}
```

Reply:
Acredito que este tipo de extensão tem potencial de criar dificuldade no debug do teste em caso de falha. Se você receber um `null` tentará descobrir qual é a origem deste nulo. Agora se receber o `JSONParseFailure` vai entender logo que está recebendo o failure ao invés do data.

Reply:
justo! apliquei os ajustes